### PR TITLE
style(init): introduce role/mode check functions

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

Remove repeated code with some small functions, let the code more readable.

Introduce new functions:
* is_data_plane 
* is_control_plane 
* is_dbless

### CODE REVIEW

File: kong/init.lua [hunk 1]
```diff
@@ -179,6 +179,32 @@ do
...
+
+
+  is_dbless = function(config)
+    return config.database == "off"
+  end
+
+
+  is_dp_worker_process = function()
+    return process.type() == "privileged agent"
+  end
```
Comment (flrgh) [inline kong/init.lua:204]:
```suggestion
  is_dp_worker_process = function(config)
    return is_data_plane(config)
           and process.type() == "privileged agent"
  end
```

Comment (chronolaw) [reply kong/init.lua:204]: No, `dp_worker` is not only for role `data_plane`, but also for role `traditional`.

Comment (chronolaw) [reply kong/init.lua:204]: According the logic before, only data_plane will enable privileged process, so I think we need not do it.

**Comment by chronolaw** [comment]
Due to #9860, this PR needs to rebase and check again.

File: kong/init.lua (hunk 1 above)
Comment (chronolaw) [reply kong/init.lua]: #9860 removed privileged process, so I also removed this function.

**Review by StarlightIbuki — APPROVED** [review]

**Pull request merged**